src/test/unit/test_user_utils.py

```diff
@@ -18,3 +18,35 @@ def test_report_error_without_enhance_print(capsys):
     report_error(msg)
     captured = capsys.readouterr()
     assert captured.out == f"RequestId: 123 {LUMIGO_REPORT_ERROR_STRING} {msg}\n"
+
+
+def test_add_tag():
+    key = "my_key"
+    value = "my_value"
+    add_tag(key, value)
+    assert SpansContainer.get_span().function_span["tags"] == [{"key": key, "value": value}]
+
+
+def test_add_tag_empty(capsys):
+    add_tag("", "value")
+    assert "Unable to add tag" in capsys.readouterr().out
+    assert SpansContainer.get_span().function_span["tags"] == []
+
+
+def test_add_tag_pass_max_chars(capsys):
+    add_tag("k" * (MAX_TAG_LEN + 1), "v" * (MAX_TAG_LEN + 1))
+    assert "Unable to add tag" in capsys.readouterr().out
```

Comment:
This is customer-facing. Print informable message!
I would move the validation to this function and print hints